src/crystal/system/win32/file.cr

```diff
@@ -26,8 +26,9 @@ module Crystal::System::File
     fd
   end
 
-  def self.mktemp(prefix : String, suffix : String?, dir : String) : {LibC::Int, String}
-    path = "#{tempdir}\\#{prefix}.#{::Random::Secure.hex}#{suffix}"
+  def self.mktemp(prefix : String?, suffix : String?, dir : String) : {LibC::Int, String}
```

Comment:
It's odd that these type restrictions are not present in the unix counter part

Reply:
I suppose, yes. I'm not changing the status quo, though.